dropwizard-zipkin/src/main/java/smartthings/dw/cassandra/DropwizardTracedSession.java

```diff
@@ -0,0 +1,55 @@
+package smartthings.dw.cassandra;
+
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.Session;
+import com.github.kristofa.brave.Brave;
+import com.google.common.base.Function;
+import com.google.common.reflect.AbstractInvocationHandler;
+import com.google.common.reflect.Reflection;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import smartthings.brave.cassandra.TraceablePreparedStatement;
+import smartthings.brave.cassandra.TracedSession;
+import zipkin.Span;
+import zipkin.reporter.Reporter;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Method;
+
+/**
+ * wrapper class to get around weirdness in
+ * smartthings.brave.cassandra.TracedSession#handleInvocation(java.lang.Object, java.lang.reflect.Method, java.lang.Object[])
+ */
+public class DropwizardTracedSession extends AbstractInvocationHandler {
+    private final Session tracedSession;
+    private final Session baseSession;
+
+    public static Session create(Session target, Brave brave, Reporter<Span> reporter, String serviceName) {
+        return Reflection.newProxy(Session.class, new DropwizardTracedSession(target, brave, reporter, serviceName));
+    }
+
+    private DropwizardTracedSession(Session target, Brave brave, Reporter<Span> reporter, String serviceName) {
+        baseSession = target;
+        tracedSession = TracedSession.create(target, brave, reporter, serviceName);
+    }
+
+    @Override
+    protected Object handleInvocation(Object proxy, Method method, Object[] args) throws Throwable {
+        if (method.getName() == "prepare") {
+            return TraceablePreparedStatement.from((PreparedStatement) method.invoke(baseSession, args), "statement");
```

Comment:
agreed that they should have a friendly name, but the spans do contain key/value like `cql.query` = `SELECT * FROM cron_schedules WHERE group = :group` in case there isn't a nice way to name accessor methods